ckanext/datastore/plugin.py

```diff
@@ -245,6 +245,7 @@ def get_actions(self):
             'datastore_info': action.datastore_info,
             'datastore_function_create': action.datastore_function_create,
             'datastore_function_delete': action.datastore_function_delete,
+            'datastore_trigger_each_row': action.datastore_trigger_each_row,
```

Comment:
Please add an IAuthFunction as well and check that the user calling the method has the same access required by datastore_upsert?